pkg/traces/config.go

```diff
@@ -978,6 +980,7 @@ func orderProcessors(processors []string, splitPipelines bool) [][]string {
 		if processor == "batch" ||
 			processor == "tail_sampling" ||
 			processor == "automatic_logging" ||
+			processor == "spanmetrics" ||
```

Comment:
Yep, that's the main bugfix. 

> if we encounter the spanmetrics processor, we order it accordingly, right?

I suppose your question is for Flow, since static mode automatically orders them? In Flow, if there are two Agents which ingest traces in a load balanced way, they need to be behind a load balancing exporter for spanmetrics to work ok.